### PR TITLE
[DCOS-29593] Use introspection of stub universe JSON to determine package name

### PR DESCRIPTION
**Pull request by mediapills**
- Added  package name loader from the stub universe URL

### CODE REVIEW

File: tools/release_builder.py [hunk 1]
```diff
@@ -44,7 +45,21 @@ def get_package_name(stub_universe_url: str) -> str:
                 )
 
             package_name = name_match.group(1)
-            log.info("Got package name %s from stub universe URL")
+
+            try:
+                log.info("Parse stub universe")
```
Comment (porridge) [inline tools/release_builder.py]:
```suggestion
                log.info("Parsing stub universe...")
```

**Comment by mediapills** [comment]
> This seems to include changes from another PR. Please make them independent.

Done 

> It's not clear to me where and how many times `get_package_name` is called. We might have to cache the stub `json` file to not fetch it every time.

It is called only once so I left it as is.

> Also, for the time being I think we should attempt to parse the file, but in case of any problem, we should fall back to the previous (filename-based) logic with a warning. Once we establish this works for all frameworks, we can phase out the old logic.

I have added a check in a case we have more than one different names `log.warn` will notify us about this.
> And probably most importantly, it's not clear to me what should happen in case there are multiple packages in the `json` file. Perhaps this is the case that's not working for DSE and the reason that repo has custom modifications around that area? Can you please investigate?

File: tools/release_builder.py [hunk 2]
```diff
@@ -44,7 +45,30 @@ def get_package_name(stub_universe_url: str) -> str:
...
+                                log.warning(
+                                    "Two different package names have been found '{}' and '{}'".format(
+                                        universe_name, package["name"]
+                                    )
+                                )
+                            else:
+                                universe_name = package["name"]
+                if universe_name is not None:
+                    package_name = universe_name
+            except urllib.error.HTTPError:
```
Comment (porridge) [inline tools/release_builder.py]: Unfortunately from my experience `urllib` is not well behaved w.r.t. exceptions, especially when TLS is involved - we cannot assume that this is the only exception type that will be raised. Please use `except Exception` to handle this gracefully.

File: tools/release_builder.py [hunk 3]
```diff
@@ -44,7 +45,30 @@ def get_package_name(stub_universe_url: str) -> str:
...
+                log.warning("Could not open URL: {}".format(stub_universe_url))
```
Comment (porridge) [inline tools/release_builder.py]: Please do not hide the details provided in the original exception. This makes troubleshooting very hard - we'd have no idea whether the problem was transient, permanent, server-side, client-side, bad proxy or what.

File: tools/release_builder.py [hunk 4]
```diff
@@ -44,7 +45,30 @@ def get_package_name(stub_universe_url: str) -> str:
...
+            try:
+                universe_name = None
+                log.info("Parsing stub universe...")
+                with urllib.request.urlopen(stub_universe_url) as response:
+                    data = json.loads(response.read())
+                    for package in data["packages"]:
+                        if "name" in package:
+                            if universe_name is not None and universe_name != package["name"]:
+                                log.warning(
+                                    "Two different package names have been found '{}' and '{}'".format(
```
Comment (porridge) [inline tools/release_builder.py]: s/Two/More than one/

**Comment by mediapills** [comment]
Could you please check the changes?

**Pull request merged**